app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -297,6 +297,22 @@ export default class TransactionStateManager extends EventEmitter {
     return transactions[txId];
   }
 
+  /**
+   * @param {string} txHash - transaction hash
+   * @returns {TransactionMeta} the txMeta who matches the given hash if none found
+   * for the network returns undefined
+   */
+  getTransactionByHash(txHash) {
+    const { transactions } = this.store.getState();
+    const txByHash = Object.values(transactions).filter((tx) => {
+      if (txHash && tx.hash && tx.hash === txHash) {
+        return tx;
+      }
+      return undefined;
+    });
+    return txByHash[0];
+  }
+
```

Comment:
Regardless of @danjm 's feedback on whether we should add the replacedById key versus using the hash, this method appears to be unused?

Reply:
@brad-decker You are right, this method is unused, I thought that I can use this method in transaction controller where I created the same method `getTransactionByHash` like this:
`_getTransactionByHash(txHash) {
    return this.txStateManager.getTransactionByHash(txHash)
  }` but I wasn't sure about that. 
What do you think about this?

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15383/commits/1c9525d3f99a0d7e073e759aa96624b20fc8c209).